File: events-notifications/subscriptions/renew-subscription-update-event-filters/code-samples/updateSubscription.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# PATH PARAMETERS
subscriptionId = '<ENTER VALUE>'

# OPTIONAL QUERY PARAMETERS
queryParams = {
    #'aggregated': true
}

# POST BODY
body = {
    'eventFilters': [
        '<ENTER VALUE>'
    ],
    'deliveryMode': {
        'transportType': 'PubNub',
        'address': '<ENTER VALUE>',
        'encryption': true,
        'certificateName': '<ENTER VALUE>',
        'registrationId': '<ENTER VALUE>',
        'verificationToken': '<ENTER VALUE>'
    },
    'expiresIn': 604800
}

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])
r = platform.put(f'/restapi/v1.0/subscription/{subscriptionId}', body, queryParams)

File: events-notifications/subscriptions/renew-subscription-update-event-filters/code-samples/updateSubscription.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$subscriptionId = '<ENTER VALUE>';

// OPTIONAL QUERY PARAMETERS
$queryParams = array(
    //'aggregated' => true
)

// POST BODY
$body = array(
    'eventFilters' => array(
        '<ENTER VALUE>'  
    ),
    'deliveryMode' => array(
        'transportType' => 'PubNub',
        'address' => '<ENTER VALUE>',
        'encryption' => true,
        'certificateName' => '<ENTER VALUE>',
        'registrationId' => '<ENTER VALUE>',
        'verificationToken' => '<ENTER VALUE>'
    ),
    'expiresIn' => 604800
);

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));
$r = $platform->put("/restapi/v1.0/subscription/{$subscriptionId}", $body, $queryParams);
// PROCESS RESPONSE
?>
